data_classes.py

```diff
@@ -0,0 +1,139 @@
+from dataclasses import dataclass, field
+from datetime import datetime
+import configs.constants as constants
+import json
+import logging
+import logging.config
+
+with open("configs/log_config.json", "r", encoding="UTF-8") as stream:
+    config = json.load(stream)
+logging.config.dictConfig(config)
+logger = logging.getLogger("standard")
+
+
+@dataclass
+class TokenSale:
+    """
+    Data of given sale transaction
+    """
+
+    sort_index: int = field(init=False)
+    avax_price_in_usd: float
+    raw_sale: dict
+    price_floor: float
+    last_sales: list
+
+    def __post_init__(self):
+        self.sort_index = self.last_sales[0]["timestamp"]
+        self.transactionID: str = self.raw_sale["id"]
+        self.contractId: str = self.raw_sale["collection"]
+        self.tokenID: int = self.raw_sale["tokenId"]
+        self.verified: str = self.raw_sale["verified"]
+        self.date: datetime = datetime.utcfromtimestamp(
+            int(self.last_sales[0]["timestamp"])
+        )
+        self.priceAVAX = round(int(self.last_sales[0]["price"]) * 10 ** (-18), 3)
+        self.priceUSD: float = round(float(self.avax_price_in_usd) * self.priceAVAX, 1)
+
+        self.isTakerAsk: bool = self.last_sales[0]["isTakerAsk"]
+
+        self.soldByFull = self.last_sales[0]["fromAddress"]
+        self.boughtByFull = self.last_sales[0]["toAddress"]
+        self.soldBy: str = self.soldByFull[:5] + "..." + self.soldByFull[-4:]
+        self.boughtBy: str = self.boughtByFull[:5] + "..." + self.boughtByFull[-4:]
+
+        if self.last_sales[0]["image"]:

```

Comment:
i think this would crash if `last_sales` is an empty array. are you guaranteed this wont be empty? if so, would be good to explain in a comment why

Reply:
yes, here https://github.com/pecio222/nft-sales-bot/pull/1/commits/77f4dceaf293fdb621f3e24a8c3a9c70e64f13b9#diff-d7e7ef9768e2c7ce5a02cd61539194b37358c4d751f4ad5935d3e308bad39357R144 will add comment